lib/shoulda/matchers/independent/delegate_method_matcher.rb

```diff
@@ -327,8 +338,21 @@ def ensure_delegate_object_has_been_specified!
           end
         end
 
+        def subject_delegates_to_nil_delegate_object_correctly?
+          return true unless delegate_object_is_nil_allowed
```

Comment:
I don't think so. I think it's clear to see that there is a conditional in this method.